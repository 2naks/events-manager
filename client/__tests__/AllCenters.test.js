import ConnectedAllCenters, { AllCenters } from '../components/AllCenters.jsx';

describe('Allcenters Component', () => {
  const initialState = {
    centerReducer: {
      status: "Error",
      message: "",
      allCenters: {
        data: [{
          id: '1',
          name: 'City Hall',
          description: {
            substr: () => {}
          },
          location: {
            substr: () => {}
          },
          capacity: "500",
          price: "50000"
        }]
      }
    }
  };

  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => str
  };

  const props = {
    dispatch: () => {},
    show: () => {},
    allCenters: {
      data: []
    }
  };

  describe('Allcenters form  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<AllCenters {...props} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render connected component', () => {
      const wrapper = mount(<Provider store={store}><ConnectedAllCenters /></Provider>);
      expect(wrapper.length).toEqual(1);
    });
  });
});
