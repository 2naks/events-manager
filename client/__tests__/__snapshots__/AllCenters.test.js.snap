// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Allcenters Component Allcenters form  should render self it should render for dumb component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllCenters
    allCenters={
      Object {
        "data": Array [],
      }
    }
    dispatch={[Function]}
    show={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavBar
          page="AllCenters"
        />,
        <div
          className="panel centers"
          id="centers"
        >
          <div
            className="container events"
            id="events"
          >
            <div
              className="row event-row"
            />
          </div>
        </div>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "page": "AllCenters",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container events"
            id="events"
          >
            <div
              className="row event-row"
            />
          </div>,
          "className": "panel centers",
          "id": "centers",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row event-row"
            />,
            "className": "container events",
            "id": "events",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "row event-row",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavBar
            page="AllCenters"
          />,
          <div
            className="panel centers"
            id="centers"
          >
            <div
              className="container events"
              id="events"
            >
              <div
                className="row event-row"
              />
            </div>
          </div>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "page": "AllCenters",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container events"
              id="events"
            >
              <div
                className="row event-row"
              />
            </div>,
            "className": "panel centers",
            "id": "centers",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row event-row"
              />,
              "className": "container events",
              "id": "events",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "row event-row",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
